lib/svgo/coa.js

```diff
@@ -300,23 +301,32 @@ function optimizeFolder(config, dir, output) {
  * @param {string} output output directory
  * @return {Promise}
  */
-function processDirectory(config, dir, files, output) {
+async function processDirectory(config, dir, files, output) {
   // take only *.svg files, recursively if necessary
   var svgFilesDescriptions = getFilesDescriptions(config, dir, files, output);
-
-  return svgFilesDescriptions.length
-    ? Promise.all(
-        svgFilesDescriptions.map((fileDescription) =>
-          optimizeFile(
-            config,
-            fileDescription.inputPath,
-            fileDescription.outputPath,
-          ),
-        ),
-      )
-    : Promise.reject(
-        new Error(`No SVG files have been found in '${dir}' directory.`),
-      );
+  
+  if ( svgFilesDescriptions.length == 0 ) {
+    throw 'No SVG files have been found in "' + dir + '" directory.';
+  }
+  
+  let results = [ ],
+      cpus = os.cpus().length;
```

Comment:
It's worth noting that since SVGO doesn't run in parallel it might not make sense to scale this by CPUs

Reply:
Eventloop run in single thread in fact. 
The use of cpu count is a number >= 1 and <= ulimit.
The better number is ulimit, but is only available under linux platform, i dont found a cross-platform way to determine max files open limit.